Makefile

```diff
@@ -24,14 +24,17 @@ build: build-linux
 
 build-macos:
 	@echo "> Building for macos..."
+	mkdir -p ./out/lifecycle
```

Comment:
I think we should build into `./out/lifecycle-<os>` dirs, to avoid surprising results or `*.exe` files in our lifecycle archives. We may also want packaging targets for different OSes but given we don't have a use case yet we can hold on that.

Reply:
Deferring to separate PR, as that issue existed previously